### PR TITLE
Feature map update

### PR DESCRIPTION
**Pull request by cadw**
I finished the map functionality, we are now able to:
- Add a Camera to the map and database by clicking
- Add a Player to the map and database by clicking
- Naming new objects by filling in the form
- Update coordinates of Camera by dragging it over the map
- Update coordinates of Player by dragging it over the map

What could we still add:
- Different kinds of players
- Color picking
- Asynchronously update the page when adding a new Object

### CODE REVIEW

File: src/main/webapp/app/map/mapview.controller.js [hunk 1]
```diff
@@ -15,33 +15,50 @@
...
+        function loadCameras () {
             Camera.query({
 
             }, onSuccess, onError);
 
             function onSuccess(data, headers) {
                 vm.cameras = data;
                 vm.queryCount = vm.totalItems;
-                drawCameras(data);
+                vm.loadPlayers(data);
```
Comment (rubenwiersma) [inline src/main/webapp/app/map/mapview.controller.js]: Kunnen de players ook los van de camera's worden geladen? Nu moeten de camera's geladen zijn om ervoor te zorgen dat de players ook laden.

File: src/main/webapp/app/map/mapview.controller.js [hunk 2]
```diff
@@ -15,33 +15,50 @@
...
 
+        function loadPlayers(cameraData) {
+            Player.query({
+
+            }, onSuccess, onError);
+
+            function onSuccess(data, headers) {
+                vm.players = data;
+                vm.queryCount = vm.totalItems;
+                drawCameras(cameraData, data);
```
Comment (rubenwiersma) [inline src/main/webapp/app/map/mapview.controller.js]: Hier ook, kunnen ze misschien beter los van elkaar worden opgeroepen? Nu is het een ketting die bij de zwakste schakel het hele tekenen van de kaart opbreekt.

File: src/main/webapp/app/map/mapview.html [hunk 3]
```diff
@@ -21,19 +21,55 @@
...
+                    <td>{{player.x}}</td>
+                    <td>{{player.y}}</td>
+                </tr>
+                </tbody>
+            </table><br />
             <h2>Diagram Legend</h2>
             <ul class="list-group">
-                <li class="list-group-item"><p style="color: #00b0ff;">Cameras</p></li>
+                <li class="list-group-item"><p style="color: blue;">Camera</p></li>
+                <li class="list-group-item"><p style="color: green;">Player</p></li>
```
Comment (ChrisBerg) [inline src/main/webapp/app/map/mapview.html]: Style tags in HTML, kan dit gefixt worden?

File: src/main/webapp/app/map/mapview.controller.js [hunk 4]
```diff
@@ -52,12 +69,26 @@
...
-                console.log("New X: " + currentCamera.x + " New Y: " + currentCamera.y);
-                Camera.update(currentCamera);
-
+                var objectType = options.target.type;
+                var objectId = options.target.id;
+
+                var currentObject;
+
+                // Determine whether to update a Camera or a Player object
+                switch(objectType) {
```
Comment (hungnguyen94) [inline src/main/webapp/app/map/mapview.controller.js]: Isn't there a better way to distinguish the objects without a switch case?

File: src/main/webapp/index.html [hunk 5]
```diff
@@ -21,200 +21,200 @@
...
-        ga('create','UA-XXXXX-X');ga('send','pageview');
-    </script>-->
+<!-- Google Analytics: uncomment and change UA-XXXXX-X to be your site's ID.
+<script>
+    (function(b,o,i,l,e,r){b.GoogleAnalyticsObject=l;b[l]||(b[l]=
+    function(){(b[l].q=b[l].q||[]).push(arguments)});b[l].l=+new Date;
+    e=o.createElement(i);r=o.getElementsByTagName(i)[0];
+    e.src='//www.google-analytics.com/analytics.js';
+    r.parentNode.insertBefore(e,r)}(window,document,'script','ga'));
+    ga('create','UA-XXXXX-X');ga('send','pageview');
```
Comment (ChrisBerg) [inline src/main/webapp/index.html]: Google Analytics die niets doen? "uncomment and change UA-XXXXX-X to be your site's ID."

**Comment by hungnguyen94** [comment]
Everything does work. I'm not sure if there's a better way to distinguish the objects on the map, but look into it.

**Comment by trombomart** [comment]
Looks fine to me! The same comments as ruben

**Comment by cadw** [comment]
Checking whether the newly created object is a player or a camera might be improved. However, keep in mind that JavaScript is not fully object-oriented, so I think the solution we have right now is fine. The inline style tags still cannot be fixed, since the CSS files are somehow overridden. I will create an issue for this.

**Pull request merged**